### PR TITLE
Delete item with relationships: configure defaults

### PR DESCRIPTION
**Pull request by benbosman**
This is a continuation of https://github.com/DSpace/DSpace/pull/2577 and should only be reviewed after merging https://github.com/DSpace/DSpace/pull/2577 in master because it contains all the code from https://github.com/DSpace/DSpace/pull/2577.

This includes support for the configuration part of https://github.com/DSpace/Rest7Contract/pull/78

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java [hunk 1]
```diff
@@ -342,6 +359,71 @@ public void delete(Context context, Relationship relationship) throws SQLExcepti
...
 
+    /**
+     * Converts virtual metadata from RelationshipMetadataValue objects to actual item metadata.
+     *
+     * @param context           The relevant DSpace context
+     * @param relationship      The relationship containing the left and right items
+     * @param copyToLeftItem    The boolean indicating whether we want to write to left item or not
+     * @param copyToRightItem   The boolean indicating whether we want to write to right item or not
+     */
+    private void copyMetadataValues(Context context, Relationship relationship, boolean copyToLeftItem,
```
Comment (paulo-graca) [inline dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java]: This PR depends on PR2577? - https://github.com/DSpace/DSpace/pull/2577/files#diff-2e89fe18482e7ac43a1a4e901a8105b7R370

Comment (paulo-graca) [reply dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java]: Didn't read the description. Sorry!

**Comment by tdonohue** [comment]
@benbosman : This PR can now be rebased on latest `master` (to make it much smaller & easier to review/test) as the related PR #2577 was just merged.  Thanks!

**Comment by benbosman** [comment]
I've replaced this with https://github.com/DSpace/DSpace/pull/2599

**Pull request closed without merging**